pyproj/geod.py

```diff
@@ -336,17 +340,265 @@ def npts(
             Latitude of the terminus point
         npts: int
             Number of points to be returned
+            (including initial and/or terminus points, if required)
         radians: bool, optional
             If True, the input data is assumed to be in radians.
+        initial_idx: int, optional
+            if initial_idx==0 then the initial point would be included in the output
+            (as the first point)
+        terminus_idx: int, optional
+            if terminus_idx==0 then the terminus point would be included in the output
+            (as the last point)
+        return_points: bool, optional
+            determines the return type of the function (see below)
 
         Returns
         -------
-        list of tuples:
-            list of (lon, lat) points along the geodesic
-            between the initial and terminus points.
+        if (return_points == True) (default):
+            list of tuples:
+                list of (lon, lat) points along the geodesic
+                between the initial and terminus points.
+        else
+            tuple of two sequences:
+                tuple of (lons, lats) of the points along the geodesic
+                lons and lats are two seqeunces with the same length
+                between the initial and terminus points.
         """
-        lons, lats = super()._npts(lon1, lat1, lon2, lat2, npts, radians=radians)
-        return list(zip(lons, lats))
+        lons, lats = super()._npts(
+            lon1,
+            lat1,
+            lon2,
+            lat2,
+            npts,
+            radians=radians,
+            initial_idx=initial_idx,
+            terminus_idx=terminus_idx,
+        )
+        return list(zip(lons, lats)) if return_points else (lons, lats)
+
+    def inv_intermediate_by_npts(
+        self,
+        out_lons: Any,
+        out_lats: Any,
+        out_azis: Any,
+        lon1: float,
+        lat1: float,
+        lon2: float,
+        lat2: float,
+        npts: int,
+        initial_idx: int = 1,
+        terminus_idx: int = 1,
+        radians: bool = False,
+    ) -> GeodIntermediateReturn:
```

Comment:
I am thinking that a namedtuple would be useful in place of the type `GeodIntermediateReturn`. This way you can store the `npts`, `del_s` `out_lats`, `out_lons`, and `out_azis` with named accessors for clarity. Here is an example [ref](https://github.com/pyproj4/pyproj/blob/2d12d5aa719ea989c8f6abdb81916d43200fe9c3/pyproj/database.pyx#L116-L261).